agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -618,3 +618,45 @@ func TestNetworkModeBridge(t *testing.T) {
 	err := networkModeTest(t, agent, "bridge")
 	require.NoError(t, err, "Networking mode bridge testing failed")
 }
+
+// TestCustomAttributesWithMaxOptions tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with maximum number of supported key, value pairs
+func TestCustomAttributesWithMaxOptions(t *testing.T) {
```

Comment:
Is there any particular reason you did this only for Linux?  The same test should work on Windows as well (though you'll need to modify the Windows version of `StartAgent` separately).